src/schemas/block.json

```
@@ -125,7 +125,7 @@
 	"BlockTag": {
 		"title": "Block tag",
 		"type": "string",
-		"enum": ["earliest", "latest", "pending"]
+		"enum": ["earliest", "latest", "pending", "finalized"]
```

```suggestion
		"enum": ["earliest", "latest", "pending", "finalized", "safe", "unsafe"]
```
I would much rather see all of them added sooner rather than later.  Delaying adding the terms doesn't give us any benefit.

I also think that we should add all of these before the merge.  If it were up to me, pre-merge they would map as follows:
```
safe = latest - 1
unsafe = latest
finalized = latest - 256
```

The idea here is to get people used to `safe` lagging a bit behind, but make it useful pre-merge so dapps can start integrating with it.  `unsafe` would map to `latest` as that is how it will map post-merge.  `finalized` being 256 blocks back (about an hour) feels reasonable to me as the timeframe is similar order of magnitude as it will be post-merge and 256 blocks back is *incredibly* unlikely to reorg.

---

IMO, we shouldn't add `safe` until we have safe block feature fully implemented. `latest - n` may not always be safe which makes `safe` unreliable and potentially dangerous.

Post-merge services will have to use `finalized` instead of `k` confirmations rule post-Merge. But before the merge values for `k` parameter vary, and I am not sure if `k = 256` is used anywhere. We may set `finalized = latest - 128` but there are services that use 16 confirmations and so forth. It makes me think that aliasing `finalized` to k-confirmation rule prior the Merge won't be useful enough for service to make a switch from one to another. I think that `finalized` should be responded with error prior the Merge, it allows to avoid any misuses of this block tag prior the Merge.

We may add `unsafe` with `latest == unsafe`, though, I am not sure if it makes sense if `safe` isn't added.

---

My problem with waiting until post merge to add these is that it leaves us in a situation where dapps cannot deploy merge-ready prior to sometime *after* the merge.  This means that there is a period of time where apps are still using `latest` or `latest - 6` even though the merge is complete and they *shouldn't* be using these values.  I can appreciate that setting these values are somewhat arbitrary prior to the merge, but I think that is outweighed by the benefit of allowing dapps to update before the switch to PoS so as soon as PoS lands they can adopt safer practices.

One option would be to allow individual Ethereum clients to configure what `finalized` and `safe` mean prior to The Merge, and then post merge these config options would be ignored.  This would allow an exchange, for example, to set `finalized` to be `latest - 6` (if that is what they want).  We can then have "reasonable defaults" for people who don't manually configure, which err on the side of safety (like `finalized = latest -256`).

---

> My problem with waiting until post merge to add these is that it leaves us in a situation where dapps cannot deploy merge-ready prior to sometime *after* the merge.

Dapps may fallback to k-confirmations rule if `finalized` has been responded with error -- this is one of the ways to get prepared pre-Merge. This may not work well for the transition period, when no execution block is yet finalized and network has left PoW. But I would expect dapps consider blockchain reads slightly prior and post Merge as potentially insecure and stop processing deposits and withdrawals for this period of time.

We may have `safe == justified_block_hash`. At least safe head algorithm starts with justified, and justified is usually an epoch closer to the head than `finalized`. Once `get_safe_head` is fully implemented, UX of the `safe` will be improved as this block will become even closer to the head.

---

Also, I am not sure about "unsafe". If we take justified as a safe block path then "unsafe" becomes an alias to "latest" - not sure if "usnafe" will ever be used unless we deprecate "latest". It should be said that using "latest" is unsafe/unreliable for most of the cases and a lot of care should be taken for one that wants to use it but not sure if "unsafe" should ever replace "latest" because the latter has different semantics meaning (the tip of the canonical chain at the moment) which is getting lost if replaced with "unsafe".

---

I do think `latest` should be deprecated and eventually (an undefined point in the distant future) removed.

Why do you think the semantic meaning is lost with `unsafe`?